openedx/openedx-platform
# [BD-13] [BB-5363] refactor: deprecates replace url related properties from ModuleSystem

## Description

Builds on the shim added by https://github.com/openedx/edx-platform/pull/29190 to deprecate the following `ModuleSystem` attributes in favour of the added ReplaceURLService:

`replace_urls`,`replace_course_urls`,`replace_jump_to_id_urls`

This change should not affect any user of edx-platform

## Supporting information

- [BD-5363](https://tasks.opencraft.com/browse/BB-5363) OpenCraft internal ticket

## Testing instructions

LMS: https://pr29880.sandbox.opencraft.hosting/
Studio: https://studio.pr29880.sandbox.opencraft.hosting/

Studio:

1. Login as a demo staff (staff@example.com / edx)
2. Open sample course [Introduction to Edx](https://studio.pr29880.sandbox.opencraft.hosting/course/course-v1:edX+ed_101+2022_02)
3. Navigate to `Section1` > `SubsectionA` > [HTML](https://studio.pr29880.sandbox.opencraft.hosting/container/block-v1:edX+ed_101+2022_02+type@vertical+block@aa464e34855a41119ab18ca30d8ed751)
4. Check the static urls in preview of HTML is correcly replaced
5. Navigate to the unit [CAPA](https://studio.pr29880.sandbox.opencraft.hosting/container/block-v1:edX+ed_101+2022_02+type@vertical+block@4f8893f02b864ff88317aa691cf11c03)
6. Click on Hint button and verify urls are correctly replaced in the hints
7. Post a course update with static url and a handout and verify that the url is correctly replaced in the preview.

LMS:

1. Login as a demo staff (staff@example.com / edx)
2. Open sample course [Introduction to Edx](https://app.pr29880.sandbox.opencraft.hosting/learning/course/course-v1:edX+ed_101+2022_02/home)
3. Verify the update and handout have correct urls
4. Click on start course
5. Verify all urls in the HTML xblock are correctly replaced.
6. Click Next to go to the CAPA unit and click on `Hint`
7. Verify the urls are correct in the hint section


Mobile API:

1. Setup mobile application using [these instructions](https://edx.readthedocs.io/projects/edx-installing-configuring-and-running/en/latest/mobile.html)
2. Login to the App and navigate to a course
3. Click on the handouts, to check the handout link is pointing to correct url and handout is accessible
